### PR TITLE
import: hash URL in paths if they are too long

### PR DESCRIPTION
**Pull request by jsynacek**
https://bugzilla.redhat.com/show_bug.cgi?id=1266775

### CODE REVIEW

**Comment by jsynacek** [comment]
```
src/import/pull-common.c:156:9: error: initializer element is not constant
         static const sd_id128_t k = SD_ID128_MAKE(df,89,16,87,01,cc,42,30,98,ab,4a,19,a6,a5,63,4f);
         ^
```

Well, this compiles just fine on my machine... Any idea how to trigger the same error?

```
$ gcc --version
gcc (GCC) 5.1.1 20150618 (Red Hat 5.1.1-4)
```

File: src/import/pull-common.c [hunk 1]
```diff
@@ -149,8 +151,21 @@ int pull_make_local_copy(const char *final, const char *image_root, const char *
...
 }
 
+static int hash_url(const char *url, char **ret) {
+        uint64_t h;
+        static const sd_id128_t k = SD_ID128_MAKE(df,89,16,87,01,cc,42,30,98,ab,4a,19,a6,a5,63,4f);
+
+        assert(url);
+
+        siphash24((uint8_t *) &h, url, strlen(url), k.bytes);
+        if (asprintf(ret, PRIx64, h) < 0)
```
Comment (zonque) [inline src/import/pull-common.c:161]: `asprintf(ret, "%" PRIx64, h)`

Comment (jsynacek) [inline src/import/pull-common.c:161]: Yep, thanks.

This won't fix the "error: initializer element is not constant" though. And I have really no idea why that won't compile on the semaphore instance. I don't know how to reproduce that in my environment, so I can only guess...

**Comment by poettering** [comment]
Use SD_ID128_ARRAY() for this case, as you need it as static initializer.

**Comment by jsynacek** [comment]
v5: https://github.com/systemd/systemd/pull/1485

**Pull request closed without merging**